cpp/vcpkg.json

```diff
@@ -37,6 +37,7 @@
     },
     "grpc",
     "gtest",
+    "libxml2",
```

Comment:
Can we use `azure-storage-files-datalake-cpp` instead of `libxml2`?

Reply:
I think this relies on updating vcpkg as discussed [here](https://github.com/apache/arrow/pull/36835#discussion_r1293105852). If we upgrade vcpkg I will update this.

Reply:
We're going to use a vcpkg installation for the manylinux builds but its going to be done in a subsequent PR.